### PR TITLE
Nodejs binding demo

### PR DESCRIPTION
**Pull request by mrrostam**

### CODE REVIEW

File: binding/nodejs/index.js [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  /**
+   * @returns the version of the Cheetah engine
+   */
+  get version() {
+    return this._version;
+  }
+
+  /**
+   * Processes a frame of audio and returns newly-transcribed text and a flag indicating if an endpoint has been detected.
+   * Upon detection of an endpoint, the client may invoke `.flush` to retrieve any remaining transcription.
```
Comment (ksyeo1010) [inline binding/nodejs/index.js:123]: `.flush()` -> so we can differentiate variables and functions

File: binding/nodejs/index.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+   *
+   * @param {Array} pcm Audio data. The audio needs to have a sample rate equal to `.sampleRate` and be 16-bit linearly-encoded.
+   * This function operates on single-channel audio. If you wish to process data in a different
+   * sample rate or format consider using `.processFile`.
```
Comment (ksyeo1010) [inline binding/nodejs/index.js:127]: I dont think we have process file?

File: binding/nodejs/index.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    const status = partialTranscriptAndStatus.status;
+    if (status !== PV_STATUS_T.SUCCESS) {
+      pvStatusToException(status, "Cheetah failed to process the audio frame");
+    }
+
+    return [ partialTranscriptAndStatus.transcript, partialTranscriptAndStatus.is_endpoint === 0 ? false : true ]
+  }
+
+  /**
+   * Processes a frame of audio and returns newly-transcribed text and a flag indicating if an endpoint has been detected.
```
Comment (ksyeo1010) [inline binding/nodejs/index.js:163]: I don't think this is correct description

File: demo/nodejs/mic.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    "-i, --audio_device_index <number>",
+    "index of audio device to use to record audio",
+    Number,
+    -1
+  )
+  .option(
+    "-e, --endpoint_duration_sec <number>",
+    "Duration of endpoint in seconds",
+    Number,
+    2
```
Comment (ksyeo1010) [inline demo/nodejs/mic.js:39]: 1 <- we decided to default to 1, I think demos also default to 1 right now.

File: binding/nodejs/index.js [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+class Cheetah {
+  /**
+   * Creates an instance of Cheetah.
+   * @param {string} accessKey AccessKey obtained from Picovoice Console (https://console.picovoice.ai/).
+   * @param {number} endpointDurationSec Duration of endpoint in seconds. A speech endpoint is detected when there is a segment of audio 
+   * (with a duration specified herein) after an utterance without any speech in it. Set to `0` to disable endpoint detection.
+   * @param {string} manualModelPath the path to the Cheetah model (.pv extension)
+   * @param {string} manualLibraryPath the path to the Cheetah dynamic library (platform-dependent extension)
+   */
+  constructor(accessKey, endpointDurationSec, manualModelPath, manualLibraryPath) {
```
Comment (ksyeo1010) [inline binding/nodejs/index.js:42]: we need to default endpointDurationSec to 1.0

File: binding/nodejs/index.js (hunk 3 above)
Comment (mrrostam) [reply binding/nodejs/index.js]: copy paste from leopard, but forgot to update the descriptions, good catch! :)

File: demo/nodejs/mic.js [hunk 6]
```diff
@@ -85,18 +87,27 @@ async function micDemo() {
...
-    process.stdout.write(partialTranscript);
-    if (isEndpoint === true) {
-      const finalTranscript = engineInstance.flush();
-      process.stdout.write(`${finalTranscript} `);
+    try {
+      const [partialTranscript, isEndpoint] = engineInstance.process(pcm);
+      process.stdout.write(partialTranscript);
+      if (isEndpoint === true) {
+        const finalTranscript = engineInstance.flush();
+        process.stdout.write(`${finalTranscript} `);
```
Comment (ksyeo1010) [inline demo/nodejs/mic.js:102]: I think its fine if we write next like after flush? If I'm not wrong this is how we do it in order terminal based demos

File: binding/nodejs/index.js [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+      this._handle === undefined
+    ) {
+      throw new PvStateError("Cheetah is not initialized");
+    }
+
+    if (pcm === undefined || pcm === null) {
+      throw new PvArgumentError(
+        `PCM array provided to 'Cheetah.process()' is undefined or null`
+      );
+    } else if (pcm.length === 0) {
```
Comment (ksyeo1010) [inline binding/nodejs/index.js:152]: I think its better pcm.length !== this._frameLength

**Review by ksyeo1010 — APPROVED** [review]

**Pull request merged**